File: ios_swiftui/TodoApp/Features/Todo/Data/Remote/TodoDTO.swift
import Foundation

// JSONPlaceholder のレスポンス型
struct TodoDTO: Decodable, Sendable {
    let id: Int
    let title: String
    let completed: Bool

    func toDomain() -> Todo {
        Todo(
            id: "api_\(id)",
            title: title,
            description: "APIから取得",
            isCompleted: completed,
            priority: .medium
        )
    }
}

File: android_compose/app/src/main/java/com/example/myapplication/features/todo/data/remote/TodoDto.kt
package com.example.myapplication.features.todo.data.remote

import com.example.myapplication.features.todo.Todo
import com.example.myapplication.features.todo.TodoPriority
import kotlinx.serialization.Serializable

@Serializable
data class TodoDto(
    val id: Int,
    val title: String,
    val completed: Boolean,
)

fun TodoDto.toEntity() = Todo(
    id = "api_$id",
    title = title,
    isCompleted = completed,
    description = "APIから取得",
    priority = TodoPriority.MEDIUM,
    createdAt = System.currentTimeMillis(),
)
